File: backend/Bugget.Entities/BO/BugBo/Bug.cs
namespace Bugget.Entities.BO;

public sealed class Bug
{
    public int? Id { get; set; }
    public int? ReportId { get; set; }
    public required string Receive { get; set; }
    public required string Expect { get; set; }
    public DateTimeOffset? CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }
    public required string CreatorUserId { get; set; }
    public int Status { get; set; } =  (int)BugStatus.Active;
    public Attachment[]? Attachments { get; set; }
    public Comment[]? Comments { get; set; }
}

File: frontend/src/types/bug.ts
import { BugStatuses } from "@/const";

import { Attachment } from "./attachment";
import { Comment } from "./comment";

export type BugClientEntity = {
  id: number;
  reportId: number;
  receive: string | null;
  expect: string | null;
  creatorUserId: string;
  createdAt: string;
  updatedAt: string;
  status: BugStatuses;
  attachments: Attachment[] | null;
  comments: Comment[] | null;
  clientId: number;
  isLocalOnly: boolean;
};

export type BugFormData = {
  receive?: string;
  expect?: string;
  status?: BugStatuses;
};

export type BugUpdateData = {
  bugId: number;
  reportId: number;
  data: BugFormData;
};

export type ResultFieldTypes = "receive" | "expect";
